mvpa2/algorithms/statis.py

```diff
@@ -0,0 +1,218 @@
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the PyMVPA package for the
+#   copyright and license terms.
+#
+### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""Transformation of individual feature spaces into a compromise common space
+
+See :class:`Statis` for more information.
+
+"""
+
+import numpy as np
+from mvpa2.base.constraints import EnsureInt, EnsureRange
+from mvpa2.base.types import is_datasetlike
+from mvpa2.base.state import ClassWithCollections
+from mvpa2.mappers.zscore import ZScoreMapper
+from mvpa2.mappers.staticprojection import StaticProjectionMapper
+from mvpa2.base.dataset import hstack
+from mvpa2.base.param import Parameter
+
+from mvpa2.base import warning
+if __debug__:
+    from mvpa2.base import debug
+
+# make only main utility visible outside
+
+__all__ = ['Statis']
+
+def normalize_ds(ds):
+    """Normalize input dataset for Statis
+
+    1) zscore each column
+    2) make sum of squares to one (Frobenius norm = 1)
+    """
+    if ds.samples.dtype == int:
+        ds.samples = ds.samples.astype('float')
```

Comment:
this function is intended to return result dataset, not kosher to modify then original dataset inplace

check for datatype should allow for various int types, e.g. use
```
if np.issubdtype(ds.samples.dtype, np.integer):
```